# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: accessible/ipc/RemoteAccessibleBase.cpp
# Commit: 3a50c2e5557f
# Full Hash: 3a50c2e5557fdd606205fe22a0af02d5d1ceb28f
# Author: James Teh <jteh@mozilla.com>
# Date: 2021-12-21 04:41:29
# Description:
#   Bug 1746827: Null check mCachedFields in RemoteAccessibleBase::Min/Cur/MaxValue/Step. r=eeejay
#   
#   It's rare that mCachedFields is null now, but it seems it can still happen.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D134268
# ==============================================================================

diff -r 4a17c3f49d0d -r 3a50c2e5557f accessible/ipc/RemoteAccessibleBase.cpp
--- a/accessible/ipc/RemoteAccessibleBase.cpp	Sat Dec 18 20:05:19 2021 +0000
+++ b/accessible/ipc/RemoteAccessibleBase.cpp	Mon Dec 20 22:52:45 2021 +0000
@@ -208,9 +208,11 @@
 
 template <class Derived>
 double RemoteAccessibleBase<Derived>::CurValue() const {
-  if (auto value = mCachedFields->GetAttribute<double>(nsGkAtoms::value)) {
-    VERIFY_CACHE(CacheDomain::Value);
-    return *value;
+  if (mCachedFields) {
+    if (auto value = mCachedFields->GetAttribute<double>(nsGkAtoms::value)) {
+      VERIFY_CACHE(CacheDomain::Value);
+      return *value;
+    }
   }
 
   return UnspecifiedNaN<double>();
@@ -218,9 +220,11 @@
 
 template <class Derived>
 double RemoteAccessibleBase<Derived>::MinValue() const {
-  if (auto min = mCachedFields->GetAttribute<double>(nsGkAtoms::min)) {
-    VERIFY_CACHE(CacheDomain::Value);
-    return *min;
+  if (mCachedFields) {
+    if (auto min = mCachedFields->GetAttribute<double>(nsGkAtoms::min)) {
+      VERIFY_CACHE(CacheDomain::Value);
+      return *min;
+    }
   }
 
   return UnspecifiedNaN<double>();
@@ -228,9 +232,11 @@
 
 template <class Derived>
 double RemoteAccessibleBase<Derived>::MaxValue() const {
-  if (auto max = mCachedFields->GetAttribute<double>(nsGkAtoms::max)) {
-    VERIFY_CACHE(CacheDomain::Value);
-    return *max;
+  if (mCachedFields) {
+    if (auto max = mCachedFields->GetAttribute<double>(nsGkAtoms::max)) {
+      VERIFY_CACHE(CacheDomain::Value);
+      return *max;
+    }
   }
 
   return UnspecifiedNaN<double>();
@@ -238,9 +244,11 @@
 
 template <class Derived>
 double RemoteAccessibleBase<Derived>::Step() const {
-  if (auto step = mCachedFields->GetAttribute<double>(nsGkAtoms::step)) {
-    VERIFY_CACHE(CacheDomain::Value);
-    return *step;
+  if (mCachedFields) {
+    if (auto step = mCachedFields->GetAttribute<double>(nsGkAtoms::step)) {
+      VERIFY_CACHE(CacheDomain::Value);
+      return *step;
+    }
   }
 
   return UnspecifiedNaN<double>();
